completions/zsh/_brew

```diff
@@ -3,154 +3,825 @@
+  _describe -t pinned-taps 'pinned taps' taps
+}
+
+__brew_any_tap() {
+  _alternative \
+    'installed-taps:installed taps:__brew_installed_taps' \
+    'official-taps:official taps:__brew_official_taps'
+}
+
+__brew_common_commands() {
+  local -a commands
+  commands=(
+    'audit:check formulae for Homebrew coding style'
+    'cat:display formula file for a formula'
+    'cleanup:uninstall unused and old versions of packages'
+    'commands:show a list of commands'
+    'config:show homebrew and system configuration'
+    'create:create a new formula'
+    'deps:list dependencies and dependants of a formula'
+    'desc:display a description of a formula'
+    'doctor:audits your installation for common issues'
+    'edit:edit a formula'
+    'fetch:download formula resources to the cache'
+    'gist-logs:generate a gist of the full build logs'
+    'home:visit the homepage of a formula or the brew project'
+    'info:information about a formula'
+    'install:install a formula'
+    'reinstall:install a formula anew; re-using its current options'
+    'leaves:show installed formulae that are not dependencies of another installed formula'
+    'link:link a formula'
+    'linkapps:symlink .app bundles provided by formulae into /Applications'
+    'list:list files in a formula or not-installed formulae'
+    'log:git commit log for a formula'
+    'missing:check all installed formuale for missing dependencies.'
+    'migrate:migrate renamed formula to new name'
+    'outdated:list formulae for which a newer version is available'
+    'pin:pin specified formulae'
+    'postinstall:perform post_install for a given formula'
+    'prune:remove dead links'
+    'remove:remove a formula'
+    'search:search for a formula (/regex/ or string)'
+    'switch:switch between different versions of a formula'
+    'tap:tap a new formula repository from GitHub, or list existing taps'
+    'tap-info:information about a tap'
+    'tap-pin:pin a tap'
+    'tap-unpin:unpin a tap'
+    'test-bot:test a formula and build a bottle'
+    'uninstall:uninstall a formula'
+    'unlink:unlink a formula'
+    'unlinkapps:remove symlinked .app bundles provided by formulae from /Applications'
+    'unpin:unpin specified formulae'
+    'untap:remove a tapped repository'
+    'update:fetch latest version of Homebrew and all formulae'
+    'upgrade:upgrade outdated formulae'
+    'uses:show formulae which depend on a formula'
+    '--cellar:brew cellar'
+    '--env:brew environment'
+    '--repository:brew repository'
+    '--version:version information'
+    '--prefix:where brew lives on this system'
+    '--cache:brew cache'
+  )
+  _describe -t common-commands 'common commands' commands
+}
+
+__brew_all_commands() {
+  local -a commands
+  commands=($(_call_program brew brew commands --quiet --include-aliases))
+  _describe -t all-commands 'all commands' commands
+}
+
+__brew_commands() {
+  # TODO remove duplicates between common and all commands
```

Comment:
I'm generally not a fan of TODOs in code, but didn't want to have this fall through the cracks / add to the complexity of this PR. This also implies some not obvious questions about whether we should show a description for every command, or instead reduce the shown commands to only the common commands (like how git does). If it's the latter, what does that list look like?